### PR TITLE
P#50177 Display Links, Ogulo-Links, Objekt-Links in the detail view 

### PR DESCRIPTION
**Pull request by hungnc89**

### CODE REVIEW

File: plugin/EstateDetail.php [hunk 1]
```diff
@@ -165,6 +204,31 @@ public function getMovieEmbedPlayers(array $options = []): array
...
+		if (
+			($type === 'ogulo' && $this->getDataView()->getOguloLinks() === LinksTypes::LINKS_EMBEDDED) ||
+			($type === 'object' && $this->getDataView()->getObjectLinks() === LinksTypes::LINKS_EMBEDDED) ||
+			($type === 'link' && $this->getDataView()->getLinks() === LinksTypes::LINKS_EMBEDDED)
+		) {
+			$links = $this->getEstateFiles()->getEstateLinks($estateId, $type);
+			$allowedOptions = array_flip(['width', 'height']);
+			$newOptions = array_intersect_key($options, $allowedOptions);
+			foreach ($links as $linkId => $properties) {
+				$player = '<iframe width="' . $newOptions['width'] . '" height="' . $newOptions['height'] . '" src="' . $properties['url'] .'" style="border: none"
```
[the remark this answers is not in the document]
Comment (hungnc89) [reply plugin/EstateDetail.php]: @jayay I tried to improve the test for EstateDetail. But it is very difficult to continue. This job will take a long time

**Comment by michaelarnold-dev** [comment]
@tung-le-esg Could you please check this?

**Comment by hungnc89** [comment]
@jayay could you please continue to review this PR?

**Comment by hungnc89** [comment]
@jayay could you please review this PR?

File: css/onoffice_defaultview.css [hunk 2]
```diff
@@ -258,6 +258,11 @@
 #oo-similarframe .oo-listobject {
 	width: inherit;
 }
+
+.oo-video .player-title {
```
Comment (jayay) [inline css/onoffice_defaultview.css:262]: Do we break existing layouts with this?

File: plugin/Types/LinksTypes.php [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+	const LINKS_EMBEDDED = 'embedded';
+
+
+	/** */
+	const FILE_TYPE_OGULO_LINK = 'Ogulo-Link';
+	const FILE_TYPE_OBJECT_LINK = 'Objekt-Link';
+	const FILE_TYPE_LINK = 'Link';
+
+	/** @var string[] */
+	private static $_linkTypes = [
```
Comment (jayay) [inline plugin/Types/LinksTypes.php:51]: Please move immutable data into a `const`.

File: css/onoffice_defaultview.css (hunk 2 above)
Comment (jayay) [reply css/onoffice_defaultview.css:262]: Same problem as https://github.com/onOfficeGmbH/oo-wp-plugin/pull/106#issuecomment-1025476719

What do you think about the possible solution?

**Comment by hoangdoegs** [comment]
@jayay Could you please review this PR?

**Review by jayay — DISMISSED** [review]

**Review by jayay — DISMISSED** [review]

**Comment by michaelarnold-dev** [comment]
Testing feedback:
Test with version 2.21.2-15-gb90e1685 on https://1.wptest.onofficeweb.com/7-2/111-schnapper/. See also the attached screenshots.

1.) Mode link: For movie links, the title is missing.
2.) Titles of links are always written in capital letters, although in enterprise are written in small letters. This has not been a requirement. They should be written like in enterprise.
3.) Mode embedded: The title of movie link and ogulo link is bigger than the other titles. Titles should have uniform size.

**Review by jmaas-onoffice — CHANGES_REQUESTED** [review]
See the requested changes in https://github.com/onOfficeGmbH/oo-wp-plugin/pull/126#issuecomment-1102692614.

**Comment by hungnc89** [comment]
- Tester tested
- Video

https://user-images.githubusercontent.com/72601287/166902026-c22a406b-e60e-42e8-8816-f6d18346f4a4.mp4

**Comment by tang-hien-egs** [comment]
@jmaas-onoffice , i fixed PR, please check it help me !

**Review by jmaas-onoffice — APPROVED** [review]

**Comment by jmaas-onoffice** [comment]
Works great. :)

I asked a colleague for feedback and they asked for some slight changes to the template. So I just implemented them myself.

**Review by jmaas-onoffice — DISMISSED** [review]

**Review by jmaas-onoffice — APPROVED** [review]

**Pull request merged**